src/tiffvisitor_int.cpp

```diff
@@ -523,9 +523,9 @@ namespace Exiv2 {
                 Exiv2::Value::AutoPtr v = Exiv2::Value::create(records[i].bSigned?Exiv2::signedShort:Exiv2::unsignedShort);
                 std::ostringstream    s;
                 if ( records[i].bSigned ) {
-                    for ( int16_t k = 0 ; k < records[i].size ; k++ ) s << " " << ints.at(nStart++);
+                    for ( uint16_t k = 0 ; k < records[i].size ; k++ ) s << " " << ints.at(nStart++);
```

Comment:
The type of [`records[i].size`](https://github.com/Exiv2/exiv2/blob/fec1813a3e7b0e4aaed7fae7ba9f9b49ca526638/src/tiffvisitor_int.cpp#L494) is `uint16_t`, so `uint16_t` is a better choice for `k`.